src/sync/installation.ts

```diff
@@ -178,6 +178,27 @@ const isBlocked = async (installationId: number): Promise<boolean> => {
 	}
 }
 
+/**
+ * Determines if an an error returned by the GitHub API means that we should retry it
+ * with a smaller request (i.e. with fewer pages).
+ * @param err the error thrown by Octokit.
+ */
+export const isRetryableWithSmallerRequest = (err): boolean => {
+	if (err.errors) {
+
+		const retryableErrors = err.errors.filter(
+			(error) => {
+				return "MAX_NODE_LIMIT_EXCEEDED" == error.type
+					|| error.message?.startsWith("Something went wrong while executing your query")
+			}
+		);
+
+		return retryableErrors.length;
+	} else {
+		return false;
+	}
+};
```

Comment:
The function `isRetryableWithSmallerRequest()` was previously the function `handleGitHubError()`. I refactored it to be more meaningful, added a test for it, and added the "Something went wrong" to the list of retryable errors.